### PR TITLE
[GLUTEN-2533][VL]Degrade arrow version to 8.0

### PR DESCRIPTION
**Pull request by JkSelf**
## What changes were proposed in this pull request?

1. Remove the arrow compile steps in Gluten.
2. Update the API change between Arrow 12 and Arrow 8 in Gluten.

## How was this patch tested?

Existing Uts

### CODE REVIEW

**Comment by rui-mo** [comment]
Below error occurs when compling with `./builddeps-veloxbe.sh`, could you take a look?

```
CMake Error at CMake/ConfigArrow.cmake:39 (message):
  Arrow Library Not Found: libarrow.so.800
Call Stack (most recent call first):
  core/CMakeLists.txt:218 (find_arrow_lib)
```

**Comment by JkSelf** [comment]
> Below error occurs when compling with `./builddeps-veloxbe.sh`, could you take a look?
> 
> ```
> CMake Error at CMake/ConfigArrow.cmake:39 (message):
>   Arrow Library Not Found: libarrow.so.800
> Call Stack (most recent call first):
>   core/CMakeLists.txt:218 (find_arrow_lib)
> ```

@rui-mo Can you try again?

**Comment by Yohahaha** [comment]
Could you add more background information about this change?

Please update Velox.md too.

**Comment by ulysses-you** [comment]
@Yohahaha  IIUC it's due to facebook velox... Facebook velox depends on arrow 8.0.0 and does not want to upgrade the version. Gluten degrades the arrow version to align with upstream so that we can use velox repo directly in future. But it's really a pity that we can not use advance arrow version.

**Comment by Yohahaha** [comment]
> @Yohahaha IIUC it's due to facebook velox... Facebook velox depends on arrow 8.0.0 and does not want to upgrade the version. Gluten degrades the arrow version to align with upstream so that we can use velox repo directly in future. But it's really a pity that we can not use advance arrow version.

Got it, thanks!

**Comment by rui-mo** [comment]
> io.glutenproject.exception.GlutenException: java.io.FileNotFoundException: libarrow.so.1200.0.0
	at io.glutenproject.vectorized.JniLibLoader$JniLoadTransaction.lambda$commit$0(JniLibLoader.java:253)

Above error was noticed in query test. Looks some change on Scala side is still needed.

**Comment by JkSelf** [comment]
> > io.glutenproject.exception.GlutenException: java.io.FileNotFoundException: libarrow.so.1200.0.0
> > at io.glutenproject.vectorized.JniLibLoader$JniLoadTransaction.lambda$commit$0(JniLibLoader.java:253)
> 
> Above error was noticed in query test. Looks some change on Scala side is still needed.

Updated. Can you try again?

**Comment by rui-mo** [comment]
@JkSelf Rebase is needed for this PR and the Velox PR.

**Comment by JkSelf** [comment]
> @JkSelf Rebase is needed for this PR and the Velox PR.

Done.

File: ep/build-velox/src/get_velox.sh [hunk 1]
```diff
@@ -16,8 +16,8 @@
 
 set -exu
 
-VELOX_REPO=https://github.com/oap-project/velox.git
-VELOX_BRANCH=update
+VELOX_REPO=https://github.com/JkSelf/velox.git
+VELOX_BRANCH=upstream
```
Comment (rui-mo) [inline ep/build-velox/src/get_velox.sh:20]: Please revert this change, thanks!
